### PR TITLE
Allow shorthand links for embedding youtube videos

### PR DESCRIPTION
**Pull request by hapurubokka**
fixes https://github.com/gamejolt/issue-tracker/issues/1347

### CODE REVIEW

File: src/_common/content/content-editor/embeds/youtube-video.ts [hunk 1]
```diff
@@ -30,23 +42,16 @@ export class YouTubeVideoEmbed extends EmbedSource {
...
 
 	isValidLink(link: string) {
-		try {
-			return new URL(link);
-		} catch (error) {
-			// Swallow error. new Url throws on invalid URLs, which can very well happen.
-			return false;
+		const results = getYoutubeVideoId(link) || false;
+
+		if (results && results.length === 11) {
```
Comment (hworld) [inline src/_common/content/content-editor/embeds/youtube-video.ts:51]: we shouldn't check the length, it might change in future

Comment (hworld) [reply src/_common/content/content-editor/embeds/youtube-video.ts:51]: Also, isValidLink should probably just return a boolean.
You can call getYoutubeVideoId above in the getLinkSource directly

File: src/app/components/forms/game/video/video.ts [hunk 2]
```diff
@@ -29,8 +30,8 @@ export default class FormGameVideo extends BaseForm<FormModel> implements FormOn
 	get videoData() {
 		const url = this.formModel._url;
 		if (url) {
-			const youtubeMatch = url.match(GameVideo.REGEX.YOUTUBE);
-			const vimeoMatch = url.match(GameVideo.REGEX.VIMEO);
+			const youtubeMatch = url.match(REGEX_YOUTUBE);
```
Comment (hworld) [inline src/app/components/forms/game/video/video.ts:33]: I feel like we can put that getYoutubeVideoId function somewhere in util folder and then change this code to use a generic function that would get the video match with the "provider". What do you think?

File: src/_common/game/video/video.model.ts [hunk 3]
```diff
@@ -6,20 +7,10 @@ export class GameVideo extends Model implements LightboxMediaModel {
...
-	// http://youtube.com/watch?v=DSvQAx5-PXU
-	// youtube.com/watch?v=DSvQAx5-PXU
-	// http://youtu.be/Y6lUVz1kdOk
-	// http://youtu.be/Y6lUVz1kdOk?testing
-	// http://vimeo.com/98hfg98dhfg
-	static readonly REGEX = {
-		VIDEO: /^(https?:\/\/)?(www\.)?(youtube\.com\/watch\?v=|youtu\.be\/|vimeo\.com\/)([a-zA-Z0-9_\-]+)(\S*)$/i,
-		YOUTUBE: /^(https?:\/\/)?(www\.)?(youtube\.com\/watch\?v=|youtu\.be\/)([a-zA-Z0-9_\-]+)(\S*)$/i,
-		VIMEO: /^(https?:\/\/)?(www\.)?(vimeo\.com\/)([a-zA-Z0-9_\-]+)(\S*)$/i,
+	readonly REGEX = {
```
Comment (hworld) [inline src/_common/game/video/video.model.ts:10]: are these used?

File: src/_common/content/content-editor/embeds/youtube-video.ts [hunk 4]
```diff
@@ -18,35 +20,25 @@ export class YouTubeVideoEmbed extends EmbedSource {
...
 	}
 
 	isValidLink(link: string) {
-		try {
-			return new URL(link);
-		} catch (error) {
-			// Swallow error. new Url throws on invalid URLs, which can very well happen.
-			return false;
-		}
+		const results = REGEX_YOUTUBE.exec(link);
```
Comment (hworld) [inline src/_common/content/content-editor/embeds/youtube-video.ts:40]: just  `return REGEX_YOUTUBE.test(link);`

**Review by hworld — CHANGES_REQUESTED** [review]

File: src/_common/game/video/video.model.ts (hunk 3 above)
Comment (hapurubokka) [reply src/_common/game/video/video.model.ts]: I can't find them being used anymore - removed.

**Review by hworld — APPROVED** [review]

**Pull request merged**